### ISSUE TITLE
Infinite Pagination in CloudWatchLogsClient.getLogEventsPaginator()

### ISSUE THREAD

**Issue by LukeHornibrookHaventec** [aws/aws-sdk-java-v2#1045]
When paginating through CloudWatch Logs events via `CloudWatchLogsClient.getLogEventsPaginator()`, the end of the log stream is not properly detected. Instead, the final page of the result set is continually requested in an infinite loop.

## Expected Behavior
The paginator should stop when the last event in the log stream has been consumed.

## Current Behavior
Once the paginator reaches the end of the log stream, the final request is repeated indefinitely.

The problem is that the GetLogEvents API signals the end of the stream by setting `nextForwardToken` or `nextBackwardToken` to the same as the requested value (and not null, like most other paginated APIs). `GetLogEventsResponseFetcher` uses `PaginatorUtils.isOutputTokenAvailable()` to detect the end of the stream, but this only looks for null or empty values.

## Possible Solution
`GetLogEventsResponseFetcher` may need a custom implementation, since `SyncPageFetcher.hasNextPage()` only provides the previous API response, and not the requested token.

As a side note, we should operate on either `nextBackwardToken` or `nextForwardToken` depending on the direction of pagination (ie `GetLogEventsRequest.startFromHead()`).

## Steps to Reproduce (for bugs)
```java
CloudWatchLogsClient.create()
	.getLogEventsPaginator(x -> {
		x.logGroupName("my-log-group");
		x.logStreamName("my-log-stream");
	})
	.events()
	.forEach(event -> {
		// this will be continually called for the final result page
	});
```

## Your Environment
* AWS Java SDK version used: 2.3.9

**Comment by varunnvs92** [comment]
I can repo the behavior. This is tricky to fix without breaking backwards compatibility as the PageFetcher interfaces only take response object. I am still deciding on the solution. I will keep you updated if there is any progress. Let me know if you have any solutions in mind.

>> As a side note, we should operate on either nextBackwardToken or nextForwardToken depending on the direction of pagination

We will not be able to support this as these auto-pagination methods are generated from [paginators](https://github.com/aws/aws-sdk-java-v2/blob/master/services/cloudwatchlogs/src/main/resources/codegen-resources/paginators-1.json#L42-L46) file. we use the `output_token` to determine the field to put the next token. As this file is based on a spec and generated from service model, it would be lot more effort to modify the spec to support backwards pagination. If we observe more methods with similar behavior, we can add this as a feature request.

**Comment by varunnvs92** [comment]
Sorry for the delay. Fixed in https://github.com/aws/aws-sdk-java-v2/pull/1057. Will be included in the next release

**Comment by varunnvs92** [comment]
Released in `2.4.6`. Let us know if there is any issues with the fix.

**Comment by LukeHornibrookHaventec** [comment]
Thanks @varunnvs92, log event pagination is looking good for me in 2.4.6.
